### ISSUE TITLE
Split indices of precomputed embeddings for reproducibility & comparison

### ISSUE THREAD

**Issue by tmadl** [mahmoodlab/HIPT#3]
Dear Team,

Excellent work - thanks for providing precomputed embeddings! 

For reproducibility, benchmarking and comparison, could you please share the indices of the train-val-test split in the patch embeddings under "embeddings_patch_lib/" - e.g. in "bcss_val_vits_tcga_brca_dino.pkl"? 

The file contains the embeddings and labels, but there seems to be no way to map them back to which case id / slide the embeddings came from, so it's difficult to use them to benchmark... Looking at patch_extraction_utils.py, the indices in the pickles seem to have something to do with "BCSS/40x/patches/summary.csv"; but that file is not provided either; so we can't reproduce the train-val-test split...

Thanks in advance!

**Comment by Richarizardd** [comment]
Hi @tmadl
- The train-val-test splits are originally defined from the original datasets. In cases in which only the training dataset was made available, I used a 10-Fold-CV with fixed random seed.
- Though not an intended feature of the current codebase, `*.pkl` files can easily modified to map back the embeddings / labels to the case_ids (will require some tweaking to `patch_extraction_utils.py`). If it helps, the order in which the images + labels are loaded are not randomized in the DataLoader. If you download the original datasets and run the script, the ordering of the iterated image filenames should align with the indices of the `*.pkl` files.
- I will look into the `summary.csv` for BCSS.

**Comment by Richarizardd** [comment]
Hi @tmadl - Apologies for the delay (was traveling earlier for CVPR).

Uploaded the splits [here](https://github.com/mahmoodlab/HIPT/blob/master/3-Self-Supervised-Eval/BCSS_split/summary.csv). You can download the BCSS dataset from the [official Grand Challenge link](https://bcsegmentation.grand-challenge.org/). For this dataset, we manually developed the train and test dataset splits and labels via 1) patching each ROI into 256-sized images, 2) assigned an image-level label to each patch using majority-voting (for multiple classes in the same image).
